src/services/Expenses.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+class Expenses {
+  constructor() {
+    this.expenses = [];
+  }
+
+  getAll() {
+    return this.expenses;
+  }
+
+  getOne(expenseId) {
+    return this.expenses.find(expense => expense.id === +expenseId);
+  }
+
+  getFiltered(searchParams) {
+    const { userId, category, from, to } = searchParams;
+
+    let filteredExpenses = this.expenses;
+
+    if (userId) {
+      filteredExpenses = filteredExpenses.filter(
+        expense => expense.userId === +userId
+      );
+    }
+
+    if (category) {
+      filteredExpenses = filteredExpenses.filter(
+        expense => expense.category === category
+      );
+    }
+
+    if (from && to) {
+      const fromDate = new Date(from);
+      const toDate = new Date(to);
+
+      filteredExpenses = filteredExpenses.filter(
+        expense => new Date(expense.spentAt) >= fromDate
+          && new Date(expense.spentAt) <= toDate
+      );
+    }
```

Comment:
can we use one filter and write all those conditions in a callback?)

Reply:
don't understand, how to do this in one filter

Reply:
```suggestion
	filteredExpenses.filter(
      expense => {a
      	if (userId && expense.userId !== +userId) {
      	  return false;
      	}
      	
      	if (category && expense.category !== category) {
      	  return false;
      	}
      	
      	// more code here
      }
    );
```

Reply:
done!